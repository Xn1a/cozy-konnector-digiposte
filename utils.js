export const getFileName = (doc) => {
  if (doc.invoice) {
    // a lot of invoices have the name Facture.pdf. I try to construct a more meaningfull and
    // unique name with invoice information
    const date = new Date(doc.invoice_data.due_on)
    return `Facture_${date.toLocaleDateString()}_${doc.invoice_data.chargeable_amount}${doc.invoice_data.currency}.pdf`
  }
  return doc.filename
}
