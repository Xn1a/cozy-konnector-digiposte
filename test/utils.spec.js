/* eslint-env jest */
import { getFileName } from '../src/utils'

describe('getFileName', () => {
  it('inserts date into bill', () => {
    expect(
      getFileName({
        invoice: true,
        invoice_data: {
          chargeable_amount: '42',
          currency: '€',
          due_on: '2017-10-09T16:12:27.704Z'
        }
      })
    ).toMatchSnapshot()
  })

  it('inserts date into account situations', () => {
    expect(
      getFileName({
        filename: 'Relevé de compte.pdf',
        creation_date: '2017-10-09T16:12:27.704Z'
      })
    ).toMatchSnapshot()
  })

  it('inserts date into file with no extensions', () => {
    expect(
      getFileName({
        filename: 'Décidément rien ne va',
        creation_date: '2017-10-09T16:12:27.704Z'
      })
    ).toMatchSnapshot()
  })

  it('returns file name when no date', () => {
    expect(
      getFileName({
        filename: 'Relevé de compte.pdf'
      })
    ).toMatchSnapshot()
  })

  it('returns file name when incorrect date', () => {
    expect(
      getFileName({
        filename: 'Relevé de compte.pdf',
        creation_date: 'Oops not a date'
      })
    ).toMatchSnapshot()
  })
})
