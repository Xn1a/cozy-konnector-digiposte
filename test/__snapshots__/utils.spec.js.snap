// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileName inserts date into account situations 1`] = `"Relevé de compte_2017-10-9.pdf"`;

exports[`getFileName inserts date into bill 1`] = `"Facture_2017-10-9_42€.pdf"`;

exports[`getFileName inserts date into file with no extensions 1`] = `"Décidément rien ne va_2017-10-9"`;

exports[`getFileName returns file name when incorrect date 1`] = `"Relevé de compte.pdf"`;

exports[`getFileName returns file name when no date 1`] = `"Relevé de compte.pdf"`;
